src/test/java/com/amazon/opendistroforelasticsearch/knn/index/KNNJNITests.java

```diff
@@ -165,7 +214,7 @@ public Void run() {
         float[] queryVector = {1.0f, 1.0f, 1.0f, 1.0f};
         String[] algoQueryParams = {"efSearch=200"};
 
-        final KNNIndex index = KNNIndex.loadIndex(indexPath, algoQueryParams);
+        final KNNIndex index = KNNIndex.loadIndex(indexPath, "l2", algoQueryParams);
         final KNNQueryResult[] results = index.queryIndex(queryVector, 30);
 
         Map<Integer, Float> scores = Arrays.stream(results).collect(
```

Comment:
Could you add test cases to ensure that invalid space names throw the expected exceptions?

Reply:
Yes. My initial thought was to add such tests at KNNSettings module b/c that is where I want the input value to be validated and fail fast. Do you think it is also necessary to prevent invalid spaceType into JNI functions?

Reply:
Added an integration test in KNNESSettingsIT.

Reply:
Makes sense thanks